packages/blade/src/components/Link/BaseLink/BaseLink.tsx

```diff
@@ -0,0 +1,271 @@
+  iconPadding: DotNotationSpacingStringToken;
+  textColor: BaseTextProps['color'];
+  focusRingColor: string;
+  motionDuration: DurationString;
+  motionEasing: EasingString;
+  cursor: CSSObject['cursor'];
+  disabled: boolean;
+  role: 'button' | 'link';
+};
+
+const getColorToken = ({
+  variant,
+  intent,
+  contrast,
+  element,
+  currentInteraction,
+  isDisabled,
+  isVisited,
+}: {
+  variant: BaseLinkProps['variant'];
+  intent: BaseLinkProps['intent'];
+  contrast: BaseLinkProps['contrast'];
+  element: 'icon' | 'text';
+  currentInteraction: keyof LinkActionStates;
+  isDisabled: boolean;
+  isVisited: boolean;
+}): IconProps['color'] | BaseTextProps['color'] => {
+  let state = currentInteraction;
+  if (isDisabled && variant == 'button') {
+    state = 'disabled';
+  }
+  if (isVisited && variant == 'anchor' && !intent) {
+    // visited state is only valid for anchor variant without any intent
+    state = 'visited';
+  }
+
+  if (intent && contrast && state !== 'visited') {
+    return `feedback.${intent}.action.${element}.link.${state}.${contrast}Contrast`;
+  }
+  return `action.${element}.link.${state}`;
+};
+
+const getProps = ({
+  theme,
+  variant,
+  currentInteraction,
+  children,
+  isDisabled,
+  intent,
+  contrast,
+  isVisited,
+}: {
+  theme: Theme;
+  variant: NonNullable<BaseLinkProps['variant']>;
+  currentInteraction: keyof LinkActionStates;
+  children?: string;
+  isDisabled: boolean;
+  intent: BaseLinkProps['intent'];
+  contrast: BaseLinkProps['contrast'];
+  isVisited: boolean;
+}): BaseLinkStyleProps => {
+  const isButton = variant === 'button';
+  const props: BaseLinkStyleProps = {
+    as: isButton ? 'button' : 'a',
+    textDecorationLine: !isButton && currentInteraction !== 'default' ? 'underline' : 'none',
+    iconColor: getColorToken({
+      variant,
+      intent,
+      contrast,
+      element: 'icon',
+      currentInteraction,
+      isDisabled,
+      isVisited,
+    }) as IconProps['color'],
+    iconPadding: !children?.trim() ? 'spacing.0' : 'spacing.1',
+    textColor: getColorToken({
+      variant,
+      intent,
+      contrast,
+      element: 'text',
+      currentInteraction,
+      isDisabled,
+      isVisited,
+    }) as BaseTextProps['color'],
+    focusRingColor: getIn(theme.colors, 'brand.primary.400'),
+    motionDuration: 'duration.2xquick',
+    motionEasing: 'easing.standard.effective',
+    cursor: isButton && isDisabled ? 'not-allowed' : 'pointer',
+    disabled: isButton && isDisabled,
+    role: isButton ? 'button' : 'link',
+  };
+
+  return props;
+};
+
+const BaseLink = ({
+  children,
+  icon: Icon,
+  iconPosition = 'left',
+  isDisabled = false,
+  onClick,
+  variant = 'anchor',
+  href,
+  target,
+  rel,
+  intent,
+  contrast = 'low',
+  accessibilityLabel,
+}: BaseLinkProps): ReactElement => {
+  const [isVisited, setIsVisited] = useState(false);
+  const { currentInteraction, setCurrentInteraction, ...syntheticEvents } = useInteraction();
```

Comment:
why do you need `useInteraction`?

Reply:
Way too much logic checks happening on multiple synthetic events and I wanted to separate it out. This could possibly be reused at some point for other components, we can move it out of the Link folder then.